src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -21,47 +71,608 @@ const ActorInit En_Box_InitVars = {
+        EnBox_SetupAction(this, EnBox_AppearAnimation);
+        this->unk_1A8 = 0;
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DEMO_KANKYO, this->dyna.actor.initPosRot.pos.x,
+                    this->dyna.actor.initPosRot.pos.y, this->dyna.actor.initPosRot.pos.z, 0, 0, 0, 0x0011);
+        Audio_PlaySoundGeneral(NA_SE_EV_TRE_BOX_APPEAR, &this->dyna.actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+    }
+}
+
+void EnBox_AppearAnimation(EnBox* this, GlobalContext* globalCtx) {
+    func_8003EC50(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+
+    if (this->unk_1A8 < 0) {
+        this->unk_1A8++;
+    } else if (this->unk_1A8 < 40) {
+        this->unk_1A8++;
+        this->dyna.actor.posRot.pos.y += 1.25f;
+    } else if (this->unk_1A8 < 60) {
+        this->alpha += 12;
+        this->unk_1A8++;
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y;
+    } else {
+        EnBox_SetupAction(this, EnBox_WaitOpen);
+    }
+}
+
+/**
+ * Chest is ready to be open
+ * something else (like, Link's actor?) probably sets unk_1F4 from outside this actor's code?
+ */
+void EnBox_WaitOpen(EnBox* this, GlobalContext* globalCtx) {
+    f32 frameCount;
+    AnimationHeader* anim;
+    s32 linkAge;
+    s32 t2;
+    Vec3f sp4C;
+    Player* player;
+
+    this->alpha = 255;
+    this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+    if (this->unk_1F4) {
+        linkAge = gSaveContext.linkAge;
+        t2 = this->unk_1F4 < 0 ? 2 : 0;
+        anim = D_809CA800[t2 + linkAge];
+        frameCount = SkelAnime_GetFrameCount(&anim->genericHeader);
+        SkelAnime_ChangeAnim(&this->skelanime, anim, 1.5f, 0, frameCount, 2, 0.0f);
+        EnBox_SetupAction(this, EnBox_Open);
+        if (this->unk_1F4 > 0) {
+            switch (this->type) {
+                case ENBOX_TYPE_SMALL:
+                case ENBOX_TYPE_6:
+                case ENBOX_TYPE_ROOM_CLEAR_SMALL:
+                case ENBOX_TYPE_SWITCH_FLAG_FALL_SMALL:
+                    break;
+                default:
+                    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_DEMO_TRE_LGT,
+                                       this->dyna.actor.posRot.pos.x, this->dyna.actor.posRot.pos.y,
+                                       this->dyna.actor.posRot.pos.z, this->dyna.actor.shape.rot.x,
+                                       this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z, 0xFFFF);
+                    func_800F5C64(0x92B);
+            }
+        }
+        osSyncPrintf("Actor_Environment_Tbox_On() %d\n", this->dyna.actor.params & 0x1F);
+        Flags_SetTreasure(globalCtx, this->dyna.actor.params & 0x1F);
+    } else {
+        player = PLAYER;
+        func_8002DBD0(&this->dyna.actor, &sp4C, &player->actor.posRot.pos);
+        if (-50.0f < sp4C.z && sp4C.z < 0.0f && fabsf(sp4C.y) < 10.0f && fabsf(sp4C.x) < 20.0f &&
```

Comment:
const on the left is something we dont have a rule on, but yeah if it were me id flip it.

and the classic parens argument of course, this is so much harder for me to read personally rofl. but yeah FeelsOkayMan